content/en/docs/pipelines/tutorials/benchmark-examples.md

```diff
@@ -0,0 +1,184 @@
+a pipeline version, and creates multiple runs simultaneously.
+
+You can specify the pipelines manifest used in the benchmark script, or you can
+use one of the preloaded samples pipelines. For example, the preloaded samples
+in Kubeflow pipelines can be used.
+Moreover, it is also a good practice to use a pipeline manifest that is representative of your particular use case. For example, if your Kubeflow Pipelines cluster is mainly used for pipelines of image recognition tasks, then it would be desirable to use an image recognition pipeline in the benchmark scripts.
+
+After a proper pipeline is chosen, the benchmark scripts will run it multiple
+times simultaneously as mentioned before. Among all the operations that the Kubeflow
+Pipelines can perform, running a pipeline is arguably the most unpredictable and
+costly one. Other operations, such as creating a pipeline (version) or creating an
+experiment, usually induce a predictable and moderate cost. For example, creating a
+pipeline version will introduce a new row in the pipeline versions table and a new
+file in minio server. The new file's size depends on the pipeline version's
+manifest. If we exclude the rare case of an extremely large manifest and assume
+an average sized manifest for each created pipeline version, the total cost of
+creating a pipeline version grows linearly with the number of pipeline versions.
+However, on the other hand, the cost of running a pipeline or a pipeline version
+involves much more uncertainty and sometimes quite a high cost. A pipeline or a
+pipeline version can have arbitrary components and hence running a pipeline or a
+pipeline version can incur arbitrary time and space complexities. For example, a step
+in a pipeline can use a customized container image which performs a super
+expensive training task. In addition, the runs in a Kubeflow Pipelines instance
+also consume more DB space than pipelines, pipeline versions, experiments, etc.
+Therefore, in order to understand the performance and scalability pain points in
+a Kubeflow Pipelines instance, it is more effective to focus on the run operation
+in workloads.
+
+## Prerequisites for running benchmark scripts
+
+To run the provided benchmark scripts, you need the following:
+
+*  A Jupyter notebook environment with access to the Kubeflow Pipelines API on
+   your Kubeflow Pipelines cluster. For example, you must be able to call the
+   CREATE, GET, DELETE, and LIST methods of the pipeline, pipeline version,
+   run, job, and experiment services from your Jupyter notebook environment.
+*  A Kubeflow Pipelines cluster. If you do not have a Kubeflow Pipelines
+   cluster, learn more about your [options for installing Kubeflow
+   Pipelines](/docs/pipelines/installation/overview/).
+*  A pipeline manifest. For example, this guide uses the
+   [taxi_updated_pool.yaml](https://storage.googleapis.com/jingzhangjz-project-pipelines/benchmarks/taxi_updated_pool.yaml)
```

Comment:
Can this pipeline manifest be published to the same location on GitHub as the benchmark notebook?

Reply:
Let me check if I can put the yaml directly in github repo. Usually, when we want to put a pipeline file to github, we put a python file (instead of a yaml, which is not very user friendly to read). However, a python file can't be used directly in upload since python file needs to be compiled first. 

A short-term alternative is that I put this yaml file to GCS storage bucket under ml-pipeline or ml-pipeline-dogfood project. 

A long-term alternative is to revise the benchmark script so it first compile a python file so a python file can be used directly with benchmark script. 

WDYT?

Reply:
SGTM

Reply:
I moved the taxi_updated_pool.yaml to our public ml-pipeline bucket at https://storage.googleapis.com/ml-pipeline/sample-benchmark/taxi_updated_pool.yaml